index.js

```diff
@@ -154,17 +154,29 @@ var MongodbDriver = Base.extend({
    * Adds an index to a collection
    *
    * @param collectionName  - The collection to add the index to
-   * @param indexName       - The name of the index to add
-   * @param columns         - The columns to add an index on
-   * @param	unique          - A boolean whether this creates a unique index
+   * @param indexOptions    - An object of options to be used as Index options
+   * @param callback
    */
-  addIndex: function(collectionName, indexName, columns, unique, callback) {
-
-    var options = {
-      indexName: indexName,
-      columns: columns,
-      unique: unique
-    };
+  addIndex: function(collectionName, indexOptions, callback) {
+    var options = {};
+    if (indexOptions.constructor.name == 'Object') {
```

Comment:
Also please always use type safe comparisions and only use non type safe where you really need to fallback.